frontend/packages/console-shared/src/utils/xterm-addon-fullscreen.ts

```diff
@@ -0,0 +1,31 @@
+import { Terminal, ITerminalAddon } from 'xterm';
+import './xterm-addon-fullscreen.scss';
+
+export class XtermAddonFullscreen implements ITerminalAddon {
+  private terminal: Terminal | undefined;
+
+  public activate(terminal: Terminal): void {
+    this.terminal = terminal;
+  }
+
+  public dispose(): void {
+    if (this.terminal && this.terminal.element.classList.contains('fullscreen')) {
```

Comment:
instead of checking for `this.terminal` here and below, will it make sense to just have

```suggestion
    if (his.terminal?.element?.classList?.contains('fullscreen')) {
```